newIDE/app/src/UI/Confirm/useConfirmDialog.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+import * as React from 'react';
+import ConfirmContext from './ConfirmContext';
+import {
+  type ShowConfirmDeleteDialogOptions,
+  type ShowConfirmDialogOptions,
+} from './ConfirmContext';
+
+const useConfirmDialog = () => {
+  const { showConfirmDialog, showConfirmDeleteDialog } = React.useContext(
+    ConfirmContext
+  );
+
+  const getConfirmation = (
+    options: ShowConfirmDialogOptions
+  ): Promise<boolean> =>
+    new Promise((resolve, reject) => {
```

Comment:
Don't write reject parameter if you don't intend to use it, to show to the reader that "this will ALWAYS resolve at some point"